scIB/metrics/highly_variable_genes.py

```diff
@@ -28,32 +28,20 @@ def hvg_overlap(adata_pre, adata_post, batch, n_hvg=500, verbose=False):
     adata_post_list = splitBatches(adata_post, batch)
     overlap = []
 
-    if ('hvg_before' in adata_pre.uns_keys()) and (set(hvg_post) == set(adata_pre.var_names)):
-        print('Using precomputed hvgs per batch')
-        hvg_pre_list = adata_pre.uns['hvg_before']
-    else:
-        hvg_pre_list = precompute_hvg_batch(adata_pre, batch, hvg_post)
-
-        for i in range(len(adata_post_list)):  # range(len(adata_pre_list)):
-            sc.pp.filter_genes(adata_post_list[i], min_cells=1)  # remove genes unexpressed (otherwise hvg might break)
-
-            # ov = list(set(adata_pre_list[i].var_names).intersection(set(hvg_pre_list[i])))
-            # adata_pre_list[i] = adata_pre_list[i][:,ov]
-            # adata_post_list[i] = adata_post_list[i][:,ov]
-            batch_var = adata_post_list[i].obs[batch][0]
-
-            n_hvg_tmp = len(hvg_pre_list[batch_var])
-            # adata_pre.uns['n_hvg'][hvg_post]#np.minimum(n_hvg, int(0.5*adata_post_list[i].n_vars))
-            if verbose:
-                print(n_hvg_tmp)
-            # if n_hvg_tmp<n_hvg:
-            #    print(adata_post_list[i].obs[batch][0]+' has less than the specified number of genes')
-            #    print('Number of genes: '+str(adata_post_list[i].n_vars))
-            # hvg_pre = sc.pp.highly_variable_genes(adata_pre_list[i], flavor='cell_ranger', n_top_genes=n_hvg_tmp, inplace=False)
-            tmp_pre = hvg_pre_list[batch_var]  # adata_pre_list[i].var.index[hvg_pre['highly_variable']]
-            hvg_post = sc.pp.highly_variable_genes(adata_post_list[i], flavor='cell_ranger', n_top_genes=n_hvg_tmp,
-                                                   inplace=False)
-            tmp_post = adata_post_list[i].var.index[hvg_post['highly_variable']]
-            n_hvg_real = np.minimum(len(tmp_pre), len(tmp_post))
-            overlap.append((len(set(tmp_pre).intersection(set(tmp_post)))) / n_hvg_real)
+    hvg_pre_list = precompute_hvg_batch(adata_pre, batch, hvg_post)
+
+    for i in range(len(adata_post_list)):  # range(len(adata_pre_list)):
+        sc.pp.filter_genes(adata_post_list[i], min_cells=1)  # remove genes unexpressed (otherwise hvg might break)
+        batch_var = adata_post_list[i].obs[batch][0]
+        n_hvg_tmp = len(hvg_pre_list[batch_var])
```

Comment:
Make sure that you are indexing correctly. In Pandas, 0 is usually assumed to be a label, not a numeric location. Also, what is batch_var supposed to represent?

I think it would be better to iterate over objects rather than indices

```suggestion
    for adata in adata_post_list:
        sc.pp.filter_genes(adata, min_cells=1)  # remove genes unexpressed (otherwise hvg might break)
        batch = adata.obs[batch].unique()[0]
        n_hvg_tmp = len(hvg_pre_list[batch])
```

Or even better, something like this

```
for batch, adata, hvgs in zip(...):
    ... 
```

Reply:
`batch_var` is just the name of the batch, it doesn't really matter which element of this column is chosen as they are all the same. I think for pandas Series the 0-th element should be fine, no?

Reply:
0 should be fine, as long as the index doesn't have a different index.

I'd still recommend simplifying the names if it's not too much trouble.